x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/transport/SecurityServerTransportInterceptor.java

```diff
@@ -140,6 +194,108 @@ public <T extends TransportResponse> void sendRequest(
         };
     }
 
+    private AsyncSender interceptForRemoteAccessRequests(final AsyncSender sender) {
+        return new AsyncSender() {
+            @Override
+            public <T extends TransportResponse> void sendRequest(
+                Transport.Connection connection,
+                String action,
+                TransportRequest request,
+                TransportRequestOptions options,
+                TransportResponseHandler<T> handler
+            ) {
+                if (shouldSendWithRemoteAccessHeaders(connection, request)) {
+                    sendWithRemoteAccessHeaders(connection, action, request, options, handler);
+                } else {
+                    // Send regular request, without remote access headers
+                    try {
+                        sender.sendRequest(connection, action, request, options, handler);
+                    } catch (Exception e) {
+                        handler.handleException(new SendRequestTransportException(connection.getNode(), action, e));
+                    }
+                }
+            }
+
+            private boolean shouldSendWithRemoteAccessHeaders(final Transport.Connection connection, final TransportRequest request) {
+                final Authentication authentication = securityContext.getAuthentication();
+                assert authentication != null : "authentication must be present in security context";
+                final Subject effectiveSubject = authentication.getEffectiveSubject();
+                final Optional<String> remoteClusterAlias = remoteClusterAliasResolver.apply(connection);
+                return remoteClusterAlias.isPresent()
+                    && remoteClusterAuthorizationResolver.resolveAuthorization(remoteClusterAlias.get()) != null
+                    && isWhitelistedForRemoteAccessHeaders(request)
+                    && effectiveSubject.getType().equals(Subject.Type.USER)
+                    && false == User.isInternal(effectiveSubject.getUser())
+                    && Arrays.stream(effectiveSubject.getUser().roles()).noneMatch(ReservedRolesStore::isReserved);
```

Comment:
This may be overly conservative but since working out how we will deal with reserved roles is still work-in-progress, it feels cleaner to explicitly fall back on the legacy model here.

Reply:
Yup, I'm fine with skipping this check -- my consideration was that a user with `superuser` role should always work but it doesn't matter much since this only applies to configured and feature flagged deployments either way.